examples/src/main/java/org/bitcoinj/examples/ForwardingService.java

```diff
@@ -70,34 +73,54 @@ public static void forward(BitcoinNetwork network, Address address) {
         System.out.println("Forwarding address: " + address);
 
         // Create the Service (and WalletKit)
-        ForwardingService forwardingService = new ForwardingService(address, network);
+        ForwardingService forwardingService = new ForwardingService(network);
 
         // Start the Service (and WalletKit)
         forwardingService.start();
 
         // Start listening and forwarding
-        forwardingService.waitForCoins();
+        CompletableFuture<Transaction> forwardedTxFuture = forwardingService.waitForCoins()
+            .thenCompose(tx -> {
+                // Incoming transaction received, now "compose" (i.e. chain) a call to wait for required confirmations
+                // The transaction "tx" can either be pending, or included into a block (we didn't see the broadcast).
+                Coin value = tx.getValueSentToMe(forwardingService.kit.wallet());
+                System.out.printf("Received tx for %s : %s\n", value.toFriendlyString(), tx);
+                System.out.println("Transaction will be forwarded after it confirms.");
+                System.out.println("Waiting for confirmation...");
+                return forwardingService.waitForConfirmation(tx);
+            })
+            .thenCompose(confidence -> {
+                // Required confirmations received, now compose a call to broadcast the forwarding transaction
+                System.out.printf("Incoming tx has received %d confirmations.\n", confidence.getDepthInBlocks());
+                // Now send the coins onwards by sending the entire contents of our wallet
+                SendRequest sendRequest = SendRequest.emptyWallet(address);
+                System.out.printf("Creating outgoing transaction for %s (calling sendTransaction()...)\n", address);
```

Comment:
Nitpick: can we perhaps remove the submessage within brackets? It seems too low-level to match the other messages.

Reply:
Yeah, sure.

Reply:
Fixed and force-pushed.